game_core/core.proto

```diff
@@ -0,0 +1,29 @@
+syntax = "proto3";
+
+package Core;
+import "types.proto";
+import "provider.proto";
+
+service GameCore {
+  // provider
+  rpc Provider(stream ProviderMsg) returns (stream ProviderMsg) {}
+
+  // client
+  rpc Login(LoginArgs) returns (LoginRet) {}
+  rpc QueryAccount(QueryAccountArgs) returns (QueryAccountRet) {}
+
+  // room
+  rpc CreateRoom(CreateRoomArgs) returns (CreateRoomRet) {}
```

Comment:
Is there any concept about room owner and some operation about release room?

Reply:
good point, I would say, release the room when the last user exits the room.
the problem is if a user disconnects, how could we detect that and release the room.
I think maybe we could add a heartbeat.
(don't want to use stream/connection,  it's really a strong restriction)